errors_test.go

```diff
@@ -56,26 +59,44 @@ func testErrorMessages(t *testing.T, filename string) {
 
 	// Write error message out as plain text.
 	var buf bytes.Buffer
-	printCompilerError(err, func(v ...interface{}) {
-		fmt.Fprintln(&buf, v...)
-	}, wd)
+	diagnostics.CreateDiagnostics(err).WriteTo(&buf, wd)
 	actual := strings.TrimRight(buf.String(), "\n")
 
 	// Check whether the error is as expected.
-	if canonicalizeErrors(actual) != canonicalizeErrors(expected) {
+	if !matchErrors(t, expected, actual) {
 		t.Errorf("expected error:\n%s\ngot:\n%s", indentText(expected, "> "), indentText(actual, "> "))
 	}
 }
 
-func canonicalizeErrors(text string) string {
-	// Fix for Windows: replace all backslashes with forward slashes so that
-	// paths will be the same as on POSIX systems.
-	// (It may also change some other backslashes, but since this is only for
-	// comparing text it should be fine).
-	if runtime.GOOS == "windows" {
-		text = strings.ReplaceAll(text, "\\", "/")
+func matchErrors(t *testing.T, pattern, actual string) bool {
+	patternLines := strings.Split(pattern, "\n")
+	actualLines := strings.Split(actual, "\n")
+	if len(patternLines) != len(actualLines) {
+		return false
 	}
-	return text
+	for i, patternLine := range patternLines {
+		indices := regexp.MustCompile(`\{\{.*?\}\}`).FindAllStringIndex(patternLine, -1)
```

Comment:
Lift the regexp compilation out of the loop?

Reply:
@aykevl want to make this change, or just merge as it is?

Reply:
If actually needed this can be done in a future PR.

Reply:
I can lift it out, but this isn't very performance critical. I don't think there's a real need (apart from aesthetics).